### PR TITLE
Add support for `std::array` in the rvs mode

### PR DESCRIPTION
**Pull request by gojakuch**
Fixes: #1000

### CODE REVIEW

File: include/clad/Differentiator/STLBuiltins.h [hunk 1]
```diff
@@ -253,6 +253,107 @@ void constructor_pullback(::std::vector<T>* v, S count, U val,
...
+  d_a->fill(0);
+}
+template <typename T, ::std::size_t N>
+void fill_pullback(::std::array<T, N>* arr,
+                   const typename ::std::array<T, N>::value_type& u,
+                   ::std::array<T, N>* d_arr,
+                   typename ::std::array<T, N>::value_type* d_u) {
+  size_t _d_i = 0;
+  size_t i = 0;
+  clad::tape<typename ::std::array<T, N>::value_type> _t1 = {};
```
[the remark this answers is not in the document]
Comment (gojakuch) [reply include/clad/Differentiator/STLBuiltins.h:296]: this is literally the declaration :)

File: include/clad/Differentiator/STLBuiltins.h [hunk 2]
```diff
@@ -253,6 +253,107 @@ void constructor_pullback(::std::vector<T>* v, S count, U val,
...
+template <typename T, ::std::size_t N>
+void fill_reverse_forw(::std::array<T, N>* a,
+                       const typename ::std::array<T, N>::value_type& u,
+                       ::std::array<T, N>* d_a,
+                       const typename ::std::array<T, N>::value_type& d_u) {
+  a->fill(u);
+  d_a->fill(0);
+}
+template <typename T, ::std::size_t N>
+void fill_pullback(::std::array<T, N>* arr,
```
Comment (parth-07) [inline include/clad/Differentiator/STLBuiltins.h:290]: Can we simplify `fill_pullback`  such that it does not use `clad::tape`? 

Also, shouldn't `fill_pullback` also update the derivative size (size of `d_arr`)?

File: include/clad/Differentiator/STLBuiltins.h [hunk 3]
```diff
@@ -253,6 +253,107 @@ void constructor_pullback(::std::vector<T>* v, S count, U val,
...
+  ::std::array<T, N> a = arr;
+  ::std::array<T, N> d_a = d_arr;
+  return {a, d_a};
+}
+template <typename T, ::std::size_t N>
+void constructor_pullback(::std::array<T, N>* a, const ::std::array<T, N>& arr,
+                          ::std::array<T, N>* d_a, ::std::array<T, N>* d_arr) {
+  for (size_t i = 0; i < N; ++i)
+    (*d_arr)[i] += (*d_a)[i];
+  // d_a->fill(0);
```
Comment (parth-07) [inline include/clad/Differentiator/STLBuiltins.h:354]: nitpick: comment can be removed.

Comment (parth-07) [reply include/clad/Differentiator/STLBuiltins.h:354]: Can we reset `d_a` size to 0 in the pullback?

File: include/clad/Differentiator/STLBuiltins.h (hunk 2 above)
Comment (gojakuch) [reply include/clad/Differentiator/STLBuiltins.h:290]: I don't think sizes of `std::array`s can be changed if I understood you correctly.

Comment (gojakuch) [reply include/clad/Differentiator/STLBuiltins.h:290]: but yeah, removing tapes is a nice idea, Vassil has also suggested that

Comment (parth-07) [reply include/clad/Differentiator/STLBuiltins.h:290]: Oh, yes, `std::array` size cannot be changed. Please ignore all of my comments related to the size of `std::array`.

Comment (gojakuch) [reply include/clad/Differentiator/STLBuiltins.h:290]: it's okay. happens :)

**Comment by gojakuch** [comment]
@vgvassilev I've fixed the failing architecture check and addressed all the review comments. I think we can merge this once the checks are completed

**Comment by vgvassilev** [comment]
Let’s ask @parth-07: is this good to go?

**Review by parth-07 — APPROVED** [review]
Looks good.

**Pull request merged**